### PR TITLE
Lexicon: votes to likes app migration

### PR DESCRIPTION
**Pull request by devinivy**
Contains the app migration to turn votes into likes.

Using the repo service here similar to #664, which takes care of updating repositories plus the generic record index. App-view related changes are handled at the end of the migrations— that would include updates to `duplicate_record` and `like` tables, in particular updating uris and cids.  A no-op `messageDispatcher` is used rather than a real one in order to avoid app-view indexing logic from kicking in.  The rkeys of the vote records are preserved into the like records.

### CODE REVIEW

File: packages/pds/src/app-migrations/votes-to-likes.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    }),
+  )
+
+  console.log(
+    SHORT_NAME,
+    `${createsTurnedDeletes.length} creates-turned-deletes`,
+    createsTurnedDeletes,
+  )
+
+  if (createsTurnedDeletes.length > voteCount / 1000) {
```
Comment (dholms) [inline packages/pds/src/app-migrations/votes-to-likes.ts:118]: is this just an arbitrary limit?

File: packages/pds/src/app-migrations/votes-to-likes.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                record,
+                validate: false, // Validated above
+              }),
+            ]
+          }),
+        )
+
+        // Chunk to avoid hitting e.g. postgres param limits
+        for (const writeChunk of chunkArray(writeResults, 1000)) {
+          await repoTx.processWrites(did, writeChunk.flat(), now)
```
Comment (dholms) [inline packages/pds/src/app-migrations/votes-to-likes.ts:99]: can't imagine it hitting this case, but maybe still worth ensuring that the chunk.length > 0

**Review by dholms — APPROVED** [review]
looks good! again, it might feel good to have a couple quick sanity checks at the end

File: packages/pds/src/app-migrations/votes-to-likes.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/app-migrations/votes-to-likes.ts]: Yeah, the idea is that if you see a non-trivial number of these that the most likely cause is that the logic to reformat the record from a vote to a like is wrong.  And if there are extremely few of them it's probably just bad data that already existed, which is fine.

**Comment by devinivy** [comment]
@dholms sweet, should be good for another peek 👍

**Review by dholms — APPROVED** [review]
yup yup yup 👍

**Pull request merged**